src/error.rs

```diff
@@ -139,14 +149,22 @@ where
         f.write_str("the conversion failed because the address of the source (a multiple of ")?;
         addr_align.fmt(f)?;
         f.write_str(") is not a multiple of the alignment (")?;
-        core::mem::align_of::<Dst>().fmt(f)?;
+        <Dst as KnownLayout>::LAYOUT.align.get().fmt(f)?;
```

Comment:
I really like the technique of using `KnownLayout` to enrich the error messages! We could also apply this to `SizeError`, I think, to revalidate the size at runtime from `src` in `Display`, rather than add another field to `SizeError`.

Reply:
Not sure I understand what you're suggesting here.

Incidentally, it turns out that some of the `Src: Deref` bounds were unnecessary (ie, not actually used), so I removed them.